### PR TITLE
Adds a new Marine gun- The Magnum

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
Adds a new marine revolver, inspired by the RU-TGMC .500 Revolver, with some sounds ~~ripped~~ borrowed from them.
Sprites by Ocelot, overall a resprite of the RU-TGMC one however.
Most stats gotten from here https://github.com/SpacelessBay/TerraGov-Marine-Corps/pull/10
Some stats by me. (Mostly ammo datum stats)

For the gun itself..
It has terrible gun handling overall, use the stock.
It has .5 windup before firing.
.75 fire delay.
It is avaliable in standard vendors.
Ammo stats are 75/40/.5 with 1 stagger, 1 knockback and no screenshake.

PASS 1
Damage increased to 100, stagger replaced with weaken (Makes it stun small xenos.)
## Why It's Good For The Game
Adds a new interesting semi-sidearm to the marine arsenal. Its basically a pocket slug thrower. Should make it interesting to use over other traditional guns in the arsenal already, and be a fun thing to shoot people with.
## Changelog
:cl:
add: Added new Marine Magnum. It has a windup to fire, staggers on hit, and deals solid damage.
/:cl:

### CODE REVIEW

**Comment by Losenis** [comment]
Does it knock marines down on FF?

**Comment by 567Turtle** [comment]
> Does it knock marines down on FF?

I fucking hope it does

**Review by Kuro020 — APPROVED** [review]

**Comment by Xsdew** [comment]
May I recommend if it isnt already the case. That the revovler is become too big to fit inside G8 pouches and backpacks/satchels.

**Comment by 567Turtle** [comment]
> May I recommend if it isnt already the case. That the revovler is become too big to fit inside G8 pouches and backpacks/satchels.

BOOOOO, I'm pretty sure the revolvers in game can be chungused already, why make it so these ones cant?

**Comment by KerbalCitizen** [comment]
>.75 fire delay.
>1 stagger
Thinking.

**Comment by SilencedMP5A5** [comment]
> > .75 fire delay.
> > 1 stagger
> 
> Thinking.

it has a windup so its probably fine

**Comment by KerbalCitizen** [comment]
> > > .75 fire delay.
> > > 1 stagger
> > 
> > 
> > Thinking.
> 
> it has a windup so its probably fine

I don't know how windup interacts with fire delay tbh. The way I imagine it is:
0 seconds: Begin windup
0.25 seconds: First shot fired
0.75 seconds: Begin windup
1.00 seconds: Second shot fired just as fire delay ends

No clue if you can begin windup during fire delay, but if you can, keeping a xeno constantly staggered with that big of a margin (1 stagger, 0.75 time between shots) seems pretty good. I believe the closest standard weapon is leicaster repeater, which has 1 stagger and 1 (10 in code, I assume 1 second?) fire delay. Then the seasonal desert eagle, which I believe is better than repeater at stagger.

**Comment by DerFlammenwerfer** [comment]
> > > > .75 fire delay.
> > > > 1 stagger
> > > 
> > > 
> > > Thinking.
> > 
> > 
> > it has a windup so its probably fine
> 
> I don't know how windup interacts with fire delay tbh. The way I imagine it is: 0 seconds: Begin windup 0.25 seconds: First shot fired 0.75 seconds: Begin windup 1.00 seconds: Second shot fired just as fire delay ends
> 
> No clue if you can begin windup during fire delay, but if you can, keeping a xeno constantly staggered with that big of a margin (1 stagger, 0.75 time between shots) seems pretty good. I believe the closest standard weapon is leicaster repeater, which has 1 stagger and 1 (10 in code, I assume 1 second?) fire delay. Then the seasonal desert eagle, which I believe is better than repeater at stagger.

The answer is no, you can't, because it will tell you the gun isn't ready to fire. You can't begin the windup during the delay.

**Comment by GreatKingWombat** [comment]
> Show the the stats like damage and pen lad

75/40

**Comment by pilotmans** [comment]
uhhh i just put the revolver in the revolver belt holster and revolver's texture is just "ERROR"

**Comment by docRoboRobert** [comment]
Windup between shots too big, recoil with full attachments and stock is too big even when not dual wielding, not worth using at all at it's current state

**Comment by Zoldorf** [comment]
Something that's relatively small that needs to be addressed, the gun can fit inside of the revolver belt even with a T-47 railscope, stock and barrel charger or extended barrel, when it should be too big to do so.

File: code/modules/projectiles/guns/revolvers.dm [hunk 1]
```diff
@@ -267,6 +267,51 @@
 	scatter = 8 // Only affects buckshot considering marksman has -15 scatter.
 	damage_falloff_mult = 1.2
 
+//-------------------------------------------------------
+// The R-76 Magnum. Fires a big round, equal to a slug. Has a windup.
+
+/obj/item/weapon/gun/revolver/standard_magnum
+	name = "\improper R-76 KC magnum"
+	desc = "The R-76 magnum is an absolute beast of a handgun used by the TGMC, rumors say it was created as a money laundering scheme by some general due to the sheer inpracticality of this firearm. Hits hard, recommended to be used with its stock attachment. Chambered in 12.7mm."
```
Comment (GreatKingWombat) [inline code/modules/projectiles/guns/revolvers.dm:275]: "The R-76 magnum is an absolute beast of a handgun used by the TGMC. Rumors say it was created as a money laundering scheme by some general due to the sheer inpracticality of this firearm. Hits hard: recommended to be used with its stock attachment. Chambered in 12.7mm."

There's more to mid-sentence punctuation than commas.

**Comment by SilencedMP5A5** [comment]
its too weak imo

File: code/modules/projectiles/gun_attachables.dm [hunk 2]
```diff
@@ -1035,6 +1035,19 @@ inaccurate. Don't worry if force is ever negative, it won't runtime.
 	desc = "A specialized stock for the BR-64."
 	icon_state = "t64stock"
 
+//You can remove the stock on the Magnum. So it has stats and is removeable.
+
+/obj/item/attachable/stock/t76
+	name = "T-76 Magnum Stock"
```
Comment (XSlayer300) [inline code/modules/projectiles/gun_attachables.dm:1041]:
```suggestion
	name = "T-76 magnum stock"
```
not on my watch

**Comment by XSlayer300** [comment]
Make sure to add the Magnum to the EORD spawn pool.

**Comment by XANAv2** [comment]
also, there is a glitch with build a sentry, it's supposed to pop the cylinder. So, you can shove a speed loader in but it doesn't, reload that way, it makes it a pain to use. This is a problem with most revolvers, and nobody has fixed it, i explained this in a ticket before this in the bug's section.

**Comment by SilencedMP5A5** [comment]
discovered a bug with shoulder mount that lets you shoot twice sometimes if you spam it during the windup

additionally, sometimes the gun just straight up stops working after you die

File: code/modules/projectiles/ammo_datums.dm [hunk 3]
```diff
@@ -479,6 +479,16 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
...
 	accuracy = -10
 
+/datum/ammo/bullet/revolver/t76
+	name = "magnum bullet"
+	handful_amount = 5
+	damage = 100
+	penetration = 40
+	sundering = 0.5
+
+/datum/ammo/bullet/revolver/t76/on_hit_mob(mob/M,obj/projectile/P)
```
Comment (Lumipharon) [inline code/modules/projectiles/ammo_datums.dm:489]:
```suggestion
/datum/ammo/bullet/revolver/t76/on_hit_mob(mob/M, obj/projectile/P)
```

File: code/modules/projectiles/guns/revolvers.dm [hunk 4]
```diff
@@ -267,6 +267,52 @@
...
+	force = 8
+	actions_types = null
+	attachable_allowed = list(
+		/obj/item/attachable/bayonet,
+		/obj/item/attachable/reddot,
+		/obj/item/attachable/flashlight,
+		/obj/item/attachable/heavy_barrel,
+		/obj/item/attachable/extended_barrel,
+		/obj/item/attachable/scope,
+		/obj/item/attachable/scope/marine,
```
Comment (Lumipharon) [inline code/modules/projectiles/guns/revolvers.dm:293]:
```suggestion
```

File: code/modules/projectiles/magazines/revolvers.dm [hunk 5]
```diff
@@ -77,3 +77,11 @@
 	default_ammo = /datum/ammo/bullet/shotgun/mbx900_buckshot
 	caliber = CALIBER_45L
 	icon_state = "h_m44"
+
+/obj/item/ammo_magazine/revolver/standard_magnum
+	name = "\improper R-76 speed loader (12.7mm)"
+	desc = "A revolver speed loader for the R-76 Magnum, mind your shoulder, will probably stagger anything it hits."
```
Comment (Lumipharon) [inline code/modules/projectiles/magazines/revolvers.dm:83]: this uh, desc is kinda misleading

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/magazines/revolvers.dm (hunk 5 above)
Comment (Pariah919) [inline code/modules/projectiles/magazines/revolvers.dm:83]:
```suggestion
	desc = "A revolver speed loader for the R-76 Magnum, mind your shoulder, will stun most moderately sized targets on impact."
```

**Review by Lumipharon — APPROVED** [review]

**Comment by chizzy376** [comment]
make it req only

**Pull request merged**